test/test_crud.py

```diff
@@ -1,16 +1,40 @@
 import pytest
+import os
+import tempfile
+import json
 from ansible.cli.playbook import PlaybookCLI
 
 
 MODULES = ['organization', 'product']
 
 
-def run_playbook(path, extra_vars=""):
-    cli = PlaybookCLI(['ansible-playbook', path, "--extra-vars", extra_vars])
-    cli.parse()
-    return cli.run()
+def run_playbook(module, record=False, extra_vars=None):
+    # Assemble parameters for playbook call
+    path = 'test/test_playbooks/{}.yml'.format(module)
+    playbook_opts = ['ansible-playbook', path, '--inventory', 'test/inventory/hosts']
+    if record:
+        # Cassettes that are to be overwritten must be deleted first
+        record_mode = 'once'
+    else:
+        # Never reach out to the internet
+        record_mode = 'none'
+        # Only run the tests (skip fixtures)
+        playbook_opts.extend(['--limit', 'tests'])
+    if extra_vars:
+        playbook_opts.extend(['--extra-vars', extra_vars])
 
+    # Dump recording parameters to json-file and pass its name by environment
+    test_params = {'test_name': module, 'serial': 0, 'record_mode': record_mode}
+    with tempfile.NamedTemporaryFile('w', suffix='json', prefix='fam-vcr-') as params_file:
+        json.dump(test_params, params_file.file)
+        params_file.file.close()
+        os.environ['FAM_TEST_VCR_PARAMS_FILE'] = params_file.name
+        # Call the playbook
```

Comment:
Everything up to this line in this function should probably just be put in a different function. #justsayin
Also, you'd be able to remove this comment if you're instead calling a function named `run_playbook`.

Reply:
Improving the structure... ACK!

Reply:
On a second thought, the assembling of the playbook parameters would be much more complicated if we split this into two functions. And the run_playbook function must be called inside the other one anyway. Right now, i cant see a way to make this really nice.

Reply:
One suggestion: move L33-L35 to a function named `run_playbook(...)` and rename this function to a more descriptive name, such as `run_vcr_playbook(...)`. :man_shrugging:

Reply:
Sorry, a minute too late...